src/pip/_internal/index.py

```diff
@@ -706,24 +706,40 @@ def egg_info_matches(
         return None
 
 
+def _parse_base_url(document, page_url):
```

Comment:
This is a nit, but I mention it because it will probably come up a bunch more times. For functions that accept parsed html, it might be better not to call the function "parse" because it has already been parsed by the time it receives it. So maybe something like `_get_base_url()` or `_determine_base_url()`. (I know I said `parse_anchor()` as an off-hand example in an earlier comment, so I'm arguing against that.)

Reply:
`_determine_base_url()` sounds good to me :)